encoding/src/parser.rs

```diff
@@ -627,9 +627,8 @@ impl<T: AsRef<[u8]>, F: CryptoFactory> EncryptedDataPayload<T, F> {
     }
 
     fn can_build_from(bytes: &[u8]) -> bool {
-        let has_acceptable_len = bytes.len() >= 12 &&
-            // TODO: Bug related to possibly insufficient number of bytes
-            fhdr_length(bytes[5]) <= bytes.len();
+        // The smallest packets contain MHDR (1 bytes) + FHDR + MIC (4 bytes).
+        let has_acceptable_len = 5 + fhdr_length(bytes[5]) <= bytes.len();
```

Comment:
Shouldn't we protect against bytes being less than 6 bytes before indexing into the 6th byte?

Reply:
Initially I thought that as there is a protection already in the phy packet parsing logic, that would be enough, but given that people might want to use this code in other context, I agree that it makes more sense to have that protection here, so I added it + a test :)